interface/billing/sl_eob_search.php

```diff
@@ -675,43 +676,41 @@ function toEncSummary(pid) {
         }
     </style>
     <?php
-    if ($GLOBALS['enable_help'] == 1) {
-        $help_icon = '<a class="pull-right oe-help-redirect" data-target="#myModal" data-toggle="modal" href="#" id="help-href" name="help-href" style="color:#676666" title="' . xla("Click to view Help") . '"><i class="fa fa-question-circle" aria-hidden="true"></i></a>';
-    } elseif ($GLOBALS['enable_help'] == 2) {
-        $help_icon = '<a class="pull-right oe-help-redirect" data-target="#myModal" data-toggle="modal" href="#" id="help-href" name="help-href" style="color:#DCD6D0 !Important" title="' . xla("Enable help in Administration > Globals > Features > Enable Help Modal") . '"><i class="fa fa-question-circle" aria-hidden="true"></i></a>';
-    } elseif ($GLOBALS['enable_help'] == 0) {
-        $help_icon = '';
+    $arrOeUiSettings = array(
+    'heading_title' => xl('EOB Posting - Search'),
+    'include_patient_name' => false,
+    'expandable' => 1,
+    'expandable_files' => array('sl_eob_search_xpd'),//all file names need suffix _xpd
+    'action' => "reset",
+    'action_title' => "",
+    'action_href' => "sl_eob_search.php",//only for actions - reset, link or back
+    'show_help_icon' => 1,
+    'help_file_name' => "sl_eob_help.php"
+    );
+    // DO NOT EDIT BELOW
+    if ($arrOeUiSettings['expandable'] && $arrOeUiSettings['expandable_files']) {
+        $arrOeUiSettings['current_state'] = collectAndOrganizeExpandSetting($arrOeUiSettings['expandable_files']);
     }
-    ?>
-    <?php
-    //to determine and set the form to open in the desired state - expanded or centered, any selection the user makes will
-    //become the user-specific default for that page. collectAndOrganizeExpandSetting() contains a single array as an
-    //argument, containing one or more elements, the name of the current file is the first element, if there are linked
-    // files they should be listed thereafter, please add _xpd suffix to the file name
-    $arr_files_php = array("sl_eob_search_xpd");
-    $current_state = collectAndOrganizeExpandSetting($arr_files_php);
-    require_once("$srcdir/expand_contract_inc.php");
+    if ($arrOeUiSettings['include_patient_name']) {
+        $arrOeUiSettings['heading_title'] .= " - " . getPatientNameFirstLast($pid);
+    }
```

Comment:
oh, this is looking nice. One last thing to really clean up this class is to migrate the above 6 lines into the constructor of the OemrUI class. (then can say bye bye to the DO NOT EDIT BELOW section)

Reply:
I'll look into this issue. We should be able to make it work :)

Reply:
The getPatientNameFirstLast and collectAndOrganizeExpandSetting should be able to work within a class (ie. I did a quick look at them to ensure they were not trying to call and out of scope variables). How did you try to call them within the class. Note $pid won't work within the class and will need to use $_SESSION['pid']

Reply:
I don't think we can get regular functions to work inside a class. If my understanding of a class being a template used to create an object is correct these functions won't work inside the template.

I tried to place it in the constructor but could not get it to work :(